modules/library/main/src/main/java/org/geotools/feature/FeatureTypes.java

```diff
@@ -344,7 +345,11 @@ public static SimpleFeatureType transform(
                 GeometryDescriptor geometryType = (GeometryDescriptor) attributeType;
 
                 tb.descriptor(geometryType);
-                if (!forceOnlyMissing || geometryType.getCoordinateReferenceSystem() == null) {
+
+                if (geometryType.getCoordinateReferenceSystem() == null
```

Comment:
I don't understand this change... looking at it, it also feels like some parenthesis may be missing to get the desired execution order (or maybe not, but parenthesis could clarify intention).

Reply:
it is correct but the second part of the disjunction could have optional parentheses. If geometryType.getCoordinateReferenceSystem() == null it cannot be checked if they are compatible (one is null).
If it isn't null, it should be checked if they are compatible but it should also be checked that the forceOnlyMissing flag is not on.